web-registry/src/lib/ecocredit/api.ts

```diff
@@ -530,10 +515,86 @@ export const queryCreditTypes = async ({
   request,
 }: QueryCreditTypesProps): Promise<QueryCreditTypesResponse> => {
   try {
-    return await client.CreditTypes({});
+    return await client.CreditTypes(request);
   } catch (err) {
     throw new Error(
       `Error in the CreditTypes query of the ledger ecocredit module: ${err}`,
     );
   }
 };
+
+// Projects
+
+interface QueryProjectsProps extends EcocreditQueryClientProps {
+  request: DeepPartial<QueryProjectsRequest>;
+}
+
+export const queryProjects = async ({
+  client,
+  request,
+}: QueryProjectsProps): Promise<QueryProjectsResponse> => {
+  try {
+    return await client.Projects(request);
+  } catch (err) {
+    throw new Error(
+      `Error in the Projects query of the ledger ecocredit module: ${err}`,
+    );
+  }
+};
+
+// queryEcoBatches consumes Regen REST endpoints - will be replaced with regen-js
+export const queryEcoBatches = async (
+  creditClassId?: string | null,
+  params?: URLSearchParams,
+): Promise<BatchInfo[]> => {
+  const client = await getQueryClient();
+  try {
+    // Currently, the experimental testnet Hambach is running an old version of regen-ledger (v2.0.0-beta1)
+    // which provides a different API than latest v4.0 for querying credit batches.
+    if (expLedger) {
+      if (creditClassId) {
+        if (!params) {
+          params = new URLSearchParams();
+        }
+        params.set('class_id', creditClassId);
+      }
+      const { data } = await axios.get(
+        `${ledgerRESTUri}/regen/ecocredit/v1alpha1/batches`,
```

Comment:
Hambach is still running V2, so we will have to leave this last reference to `v1alpha1`